modules/social_features/social_group/modules/social_group_flexible_group/social_group_flexible_group.module

```diff
@@ -694,70 +694,37 @@ function social_group_flexible_group_views_query_alter(ViewExecutable $view, Que
   // Otherwise views doesn't accept it.
   $query->addTag('flexible_group_by');
 
-  // AN users can only see flexible groups that are public.
-  if ($account->isAnonymous()) {
-    // Add context so for AN it will have a different cache.
-    $view->element['#cache']['contexts'][] = 'user.roles:anonymous';
-    $query->setGroupOperator('OR');
-    // Secret group also alters the query, so lets do it better.
-    if (\Drupal::moduleHandler()->moduleExists('social_group_secret')) {
-      $query->addWhere(1, 'groups_field_data.type', ['flexible_group'], 'NOT IN');
-      $query->addWhere(2, 'groups_field_data.type', [
-        'flexible_group',
-        'secret_group',
-        'closed_group',
-        'open_group',
-      ], 'NOT IN');
-      $query->addWhere($new_where, 'field_flexible_group_visibility_value', ['public'], 'IN');
-      $query->addWhere($new_where, 'groups_field_data.type', ['flexible_group'], 'IN');
-      return;
-    }
+  // Get all LU groups.
+  $my_groups = \Drupal::service('social_group.helper_service')
+    ->getAllGroupsForUser($account->id());
 
-    // Make sure we remove flexible group as an option so only public groups
-    // are part of this clause.
-    $query->addWhere($current_where, 'groups_field_data.type', ['flexible_group'], 'NOT IN');
+  // Get all hidden groups.
+  $hidden_groups = \Drupal::entityTypeManager()->getStorage('group')->loadByProperties([
+    'field_flexible_group_visibility' => 'members',
+  ]);
+
+  // Get all hidden groups that the current user is not a member of
+  // and remove them from showing in the view.
+  $ids = array_diff(array_keys($hidden_groups), $my_groups);
+  if ($ids) {
+    // Anonymous user should only see 'public' groups.
+    if ($account->isAnonymous()) {
+      $community_groups = \Drupal::entityTypeManager()->getStorage('group')->loadByProperties([
+        'field_flexible_group_visibility' => 'community',
+      ]);
+      $ids = array_merge($ids, array_keys($community_groups));
+
+      // Add context so for AN it will have a different cache.
+      $view->element['#cache']['contexts'][] = 'user.roles:anonymous';
+    }
 
-    // OR it is a flexible group but than we only want groups
-    // that have the public content visibility for AN users.
-    $query->setWhereGroup('AND', $new_where);
-    $query->addWhere($new_where, 'field_flexible_group_visibility_value', ['public'], 'IN');
-    $query->addWhere($new_where, 'groups_field_data.type', ['flexible_group'], 'IN');
-    return;
+    $query->addWhere($new_where, 'id', $ids, 'NOT IN');
```

Comment:
Think we need to change this to 
$query->addWhere($new_where, 'groups_field_data.id', $ids, 'NOT IN');

Because this query is making a lot joins, and ends up in this id being ambiguous.